extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/WebSocketServerConnection.java

```diff
@@ -0,0 +1,171 @@
+import io.smallrye.mutiny.Uni;
+
+/**
+ * This interface represents a client connected to a specific {@link WebSocket} endpoint.
+ * <p>
+ * Quarkus provides a built-in CDI bean of type {@code WebSocketServerConnection} that can be injected in a {@link WebSocket}
+ * endpoint and used to interact with the connected client, or all clients connected to the endpoint respectively
+ * (broadcasting).
+ * <p>
+ * Specifically, it is possible to send messages using blocking and non-blocking methods, declared on
+ * {@link BlockingSender} and {@link Sender} respectively.
+ */
+@Experimental("This API is experimental and may change in the future")
+public interface WebSocketServerConnection extends Sender, BlockingSender {
+
+    /**
+     *
+     * @return the unique identifier assigned to this connection
+     */
+    String id();
+
+    /**
+     *
+     * @param name
+     * @return the actual value of the path parameter or null
+     * @see WebSocket#path()
+     */
+    String pathParam(String name);
+
+    /**
+     * Sends messages to all open clients connected to the same WebSocket endpoint.
+     *
+     * @return the broadcast sender
+     * @see #getOpenConnections()
+     */
+    BroadcastSender broadcast();
+
+    /**
+     * Sends messages to all open clients connected to the same WebSocket endpoint and matching the given filter predicate.
+     *
+     * @param filter
+     * @return the broadcast sender
+     * @see #getOpenConnections()
+     */
+    BroadcastSender broadcast(Predicate<WebSocketServerConnection> filter);
+
+    /**
+     * The returned set also includes the connection this method is called upon.
+     *
+     * @return the set of open connections to the same endpoint
+     */
+    Set<WebSocketServerConnection> getOpenConnections();
+
+    /**
+     * @return {@code true} if the HTTP connection is encrypted via SSL/TLS
+     */
+    boolean isSecure();
+
+    /**
+     * @return {@code true} if the WebSocket is closed
+     */
+    boolean isClosed();
+
+    /**
+     *
+     * @return {@code true} if the WebSocket is open
+     */
+    default boolean isOpen() {
+        return !isClosed();
+    }
+
+    /**
+     * Close the connection.
+     *
+     * @return a new {@link Uni} with a {@code null} item
+     */
+    @CheckReturnValue
+    Uni<Void> close();
+
+    /**
+     * Close the connection.
+     */
+    default void closeAndAwait() {
+        close().await().indefinitely();
+    }
+
+    /**
+     *
+     * @return the handshake request
+     */
+    HandshakeRequest handshakeRequest();
+
+    /**
+     * Makes it possible to send messages to all clients connected to the same WebSocket endpoint.
+     *
+     * @see WebSocketServerConnection#getOpenConnections()
+     */
+    interface BroadcastSender extends Sender, BlockingSender {
+
+    }
+
+    /**
+     * Provides some useful information about the initial handshake request.
+     */
+    interface HandshakeRequest {
```

Comment:
These 2 interfaces should IMHO be top-level.

Reply:
Why? I mean, they only make sense in the context of websocket connection. In any case, we can discuss this later. It's not a high priority.

Reply:
Everything here only makes sense in the context of a websocket connection :-) I mean, if `Sender` and `BlockingSender` are top-level, surely `BroadcastSender` should be top-level. It is less clear in case of the `HandshakeRequest`, that makes sense as a nested class too.

Reply:
> Everything here only makes sense in the context of a websocket connection :-) 

More or less ;-)

> I mean, if Sender and BlockingSender are top-level, surely BroadcastSender should be top-level.

That's disputable. The connection implements `Sender` and in theory there may be other impls around. Whereas broadcasting is specific for the connection.

But like I said we can discuss it later. It does not really matter in this phase.

Reply:
If it feels like I'm insisting, that's because I know that the SI unit of temporariness is "one forever" :laughing:

Reply:
I understand your concerns but TBH if this should be biggest problem.. I can live with that ;-)